modulemd/v2/modulemd-translation-entry.c

```diff
@@ -0,0 +1,623 @@
+
+  g_hash_table_unref (te->profile_descriptions);
+  te->profile_descriptions =
+    _modulemd_hash_table_deep_str_copy (self->profile_descriptions);
+
+  return g_object_ref (te);
+}
+
+
+static void
+modulemd_translation_entry_finalize (GObject *object)
+{
+  ModulemdTranslationEntry *self = (ModulemdTranslationEntry *)object;
+
+  g_clear_pointer (&self->locale, g_free);
+  g_clear_pointer (&self->summary, g_free);
+  g_clear_pointer (&self->description, g_free);
+
+  G_OBJECT_CLASS (modulemd_translation_entry_parent_class)->finalize (object);
+}
+
+
+void
+modulemd_translation_entry_set_summary (ModulemdTranslationEntry *self,
+                                        const gchar *summary)
+{
+  g_return_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self));
+
+  g_clear_pointer (&self->summary, g_free);
+  self->summary = g_strdup (summary);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_SUMMARY]);
+}
+
+
+const gchar *
+modulemd_translation_entry_get_summary (ModulemdTranslationEntry *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self), NULL);
+
+  return self->summary;
+}
+
+
+void
+modulemd_translation_entry_set_description (ModulemdTranslationEntry *self,
+                                            const gchar *description)
+{
+  g_return_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self));
+
+  g_clear_pointer (&self->description, g_free);
+  self->description = g_strdup (description);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_DESCRIPTION]);
+}
+
+
+const gchar *
+modulemd_translation_entry_get_description (ModulemdTranslationEntry *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self), NULL);
+
+  return self->description;
+}
+
+
+static void
+modulemd_translation_entry_set_locale (ModulemdTranslationEntry *self,
+                                       const gchar *locale)
+{
+  g_return_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self));
+
+  /* It is a coding error if we ever get a NULL name here */
+  g_return_if_fail (locale);
+
+  /* It is a coding error if we ever get the default string here */
+  g_return_if_fail (g_strcmp0 (locale, TE_DEFAULT_STRING));
+
+  g_clear_pointer (&self->locale, g_free);
+  self->locale = g_strdup (locale);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_LOCALE]);
+}
+
+
+const gchar *
+modulemd_translation_entry_get_locale (ModulemdTranslationEntry *self)
+{
+  g_return_val_if_fail (MODULEMD_IS_TRANSLATION_ENTRY (self), NULL);
+
+  return self->locale;
+}
+
+
+gchar **
+modulemd_translation_entry_get_profiles_as_strv (
+  ModulemdTranslationEntry *self)
+{
+  GPtrArray *keys = _modulemd_ordered_str_keys (self->profile_descriptions,
+                                                _modulemd_strcmp_sort);
+  // Add the NULL sentinel
+  g_ptr_array_add (keys, NULL);
+  // Store the pdata for returning after we free the container
+  gchar **result = (gchar **)keys->pdata;
+  g_ptr_array_free (keys, FALSE);
```

Comment:
Could you make this conversion into a utility function, please? It's probably going to be frequently handy.